data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/index/NoIsmPolicyManagement.java

```diff
@@ -40,7 +41,7 @@ public Optional<String> checkAndCreatePolicy() throws IOException {
     @Override
     public List<String> getIndexPatterns(final String indexAlias) {
         checkArgument(StringUtils.isNotEmpty(indexAlias));
-        return Collections.singletonList(indexAlias);
+        return Collections.singletonList(indexAlias.replaceAll(TIME_PATTERN_REGULAR_EXPRESSION, "*"));
```

Comment:
This index pattern gets used by the the index template. If I understand correctly, even when there's no ISM policy, we still need wildcards for the time pattern. 

Previously, if template file is given but ISM policy file is not, the index pattern will not have `*` in it. This will lead to a bug that the index template will not be applied to the index if the index name has time pattern in it.